astropy/stats/src/fast_sigma_clip.c

```diff
@@ -105,6 +105,8 @@ static void _sigma_clip_fast(
     double *data_buffer = NULL;
     double *mad_buffer = NULL;
 
+    NPY_BEGIN_THREADS_DEF
+
     // data_buffer is used to store the current values being sigma clipped
     data_buffer = (double *)PyArray_malloc(n_i * sizeof(double));
```

Comment:
@mhvk - since you wrote this code originally, is there any reason we can't use a plain malloc here? (which would allow the whole block to be multi-threaded.

Reply:
I used it because I thought that since it came from numpy it would perhaps do something special. If it is causing problems, I don't see a reason not to replace with `malloc`. In addition, perhaps raise an issue over at numpy just to get feedback?

Reply:
`PyArray_malloc` should be safe to call without the GIL (https://github.com/numpy/numpy/pull/7404/files), it points to `PyMem_RawMalloc` which is a malloc wrapper allowing to track allocations with tracemalloc.